src/admin_upload/views.py

```diff
@@ -0,0 +1,89 @@
+from django.shortcuts import render, redirect
+from django.http import HttpResponse, JsonResponse
+from django.contrib.auth.decorators import user_passes_test
+from geonode.layers.models import Dataset
+import json
+import psycopg2
+import sys
+from woolpert import settings
+
+def admin_required(function):
+    def wrap(request, *args, **kwargs):
+        user = request.user
+        if user.is_superuser:
+            return function(request, *args, *kwargs)
+        else:
+            return redirect("/")
+    wrap.__doc__ = function.__doc__
+    wrap.__name__ = function.__name__
+    return wrap
+
+# Create your views here.
+@admin_required
+def admin_form(request):
+    if(request.POST):
+        json_data = json.loads(request.POST.get('json_data'))
+        x_index = [i for i, x in enumerate(json_data["columns"]) if x == 'x_dd']
+        y_index = [i for i, x in enumerate(json_data["columns"]) if x == 'y_dd']
+
+        dbHost     = settings.DATABASE_HOST
+        dbName  = settings.GEONODE_GEODATABASE
+        dbPassword = settings.GEONODE_GEODATABASE_PASSWORD
+        dbUsername    = settings.GEONODE_GEODATABASE_USER
+
+        error_list = []
+
+        conn = psycopg2.connect(
+            database=dbName, user=dbUsername, password=dbPassword, host=dbHost, port="5432"
+        )
+        cursor = conn.cursor()
+        cursor.execute(f"SELECT max(fid) as max_id FROM {json_data['layer']}")
+        max_id = cursor.fetchall()
+
+        cursor.execute(f"SELECT * FROM {json_data['layer']}")
+        data = cursor.fetchall()
+        # cursor.execute("select oid,typname from pg_type;")
+        # list_type = cursor.fetchall()
+        # plus two for id and location
+        column_len = len(json_data['columns']) + 2
+        print(f"col {column_len} db {len(cursor.description)}")
+
+        if len(cursor.description) != column_len:
```

Comment:
Nice catch, since users will have a template to use to populate data. The columns will definitely always match. But lets keep it here